internal/blocktx/integration_test/reorg_integration_test.go

```diff
@@ -0,0 +1,210 @@
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/bitcoin-sv/arc/internal/blocktx"
+	"github.com/bitcoin-sv/arc/internal/blocktx/blocktx_api"
+	"github.com/bitcoin-sv/arc/internal/blocktx/store/postgresql"
+	"github.com/golang-migrate/migrate/v4"
+	migratepostgres "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	_ "github.com/lib/pq"
+	"github.com/libsv/go-p2p"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/libsv/go-p2p/wire"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	postgresPort   = "5432"
+	migrationsPath = "file://../store/postgresql/migrations"
+	dbName         = "main_test"
+	dbUsername     = "arcuser"
+	dbPassword     = "arcpass"
+)
+
+var (
+	dbInfo string
+	dbConn *sql.DB
+)
+
+func TestMain(m *testing.M) {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("failed to create pool: %v", err)
+	}
+
+	port := "5436"
+	opts := dockertest.RunOptions{
+		Repository: "postgres",
+		Tag:        "15.4",
+		Env: []string{
+			fmt.Sprintf("POSTGRES_PASSWORD=%s", dbPassword),
+			fmt.Sprintf("POSTGRES_USER=%s", dbUsername),
+			fmt.Sprintf("POSTGRES_DB=%s", dbName),
+			"listen_addresses = '*'",
+		},
+		ExposedPorts: []string{"5432"},
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			postgresPort: {
+				{HostIP: "0.0.0.0", HostPort: port},
+			},
+		},
+	}
+
+	resource, err := pool.RunWithOptions(&opts, func(config *docker.HostConfig) {
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{
+			Name: "no",
+		}
+		config.Tmpfs = map[string]string{
+			"/var/lib/postgresql/data": "",
+		}
+	})
+	if err != nil {
+		log.Fatalf("failed to create resource: %v", err)
+	}
+
+	hostPort := resource.GetPort("5432/tcp")
+
+	dbInfo = fmt.Sprintf("host=localhost port=%s user=%s password=%s dbname=%s sslmode=disable", hostPort, dbUsername, dbPassword, dbName)
+	dbConn, err = sql.Open("postgres", dbInfo)
+	if err != nil {
+		log.Fatalf("failed to create db connection: %v", err)
+	}
+	err = pool.Retry(func() error {
+		return dbConn.Ping()
+	})
+	if err != nil {
+		log.Fatalf("failed to connect to docker: %s", err)
+	}
+
+	driver, err := migratepostgres.WithInstance(dbConn, &migratepostgres.Config{
+		MigrationsTable: "metamorph",
+	})
+	if err != nil {
+		log.Fatalf("failed to create driver: %v", err)
+	}
+
+	migrations, err := migrate.NewWithDatabaseInstance(migrationsPath, "postgres", driver)
+	if err != nil {
+		log.Fatalf("failed to initialize migrate instance: %v", err)
+	}
+	err = migrations.Up()
+	if err != nil && !errors.Is(err, migrate.ErrNoChange) {
+		log.Fatalf("failed to initialize migrate instance: %v", err)
+	}
+
+	code := m.Run()
+
+	err = pool.Purge(resource)
+	if err != nil {
+		log.Fatalf("failed to purge pool: %v", err)
+	}
+
+	os.Exit(code)
+}
+
+func TestReorg(t *testing.T) {
```

Comment:
This this test really Reorg?

I think it is testing just a new block being received at the same hight, but smaller chain work, not?

Reply:
Well, the idea is for it to become the Reorg test in the near future (as per the next tasks described in the top). It's just that in this PR we don't perform reorg yet. So this test just checks if the first incoming competing block with smaller chainwork will receive status `STALE` and then the next incoming competing block with a higher chainwork will receive status `LONGEST`.

But I changed the name to `TestBlockStatus` to be more clear, and will rename it back to `TestReorg` when we have reorg implemented. I hope that's okay :)

Reply:
Alright, nice 👍